progetto/ServerProject/src/server/AccountListMonitor.java

```diff
@@ -1,287 +1,292 @@
+                accountList[lastFreeposition] = account;
                 account.setAccountId(lastFreeposition);
-                posizione=lastFreeposition;//System.err.println(" cache!");
-                lastFreeposition=-1;
+                posizione = lastFreeposition;//System.err.println(" cache!");
+                lastFreeposition = -1;
                 return posizione;
-            }else{
-               for (int i=0;i<this.MAXACCOUNTNUMBER;i++){
-                   if(accountList[i]==null){
-                       accountList[i]=account;
-                       account.setAccountId(i);
-                       return i;
-                   }
-               }
+            } else {
+                for (int i = 0; i < this.MAXACCOUNTNUMBER; i++) {
+                    if (accountList[i] == null) {
+                        accountList[i] = account;
+                        account.setAccountId(i);
+                        return i;
+                    }
+                }
             }
-        }finally{
+        } finally {
             this.length++;
             listLock.writeLock().unlock();
         }
         throw new AccountMonitorRuntimeException("ERRORE:addAccount");//Non dovrebbe mai essere sollevata :D speremmu!
     }
 
-    public Account getAccountCopy(int accountId){
+    public Account addAccount(Account account, int accountId) {
+        testRange(accountId);
 
-        Account snapShot;
-        Account curr;
-
-        if(accountId>=this.MAXACCOUNTNUMBER||accountId<0){
-            throw new IllegalArgumentException("accountId>MAXACCOUNTNUMBER || accountId<0");
-        }
-
-        this.listLock.readLock().lock();
-        try {
-            curr = accountList[accountId];
-            if(curr==null){
-                return null;
-            }
-            snapShot = new
-                    Account(curr.getUsername(), curr.getPassword(), curr.getStub(), curr.getPublicKey(), curr.getAccountId());
-            return snapShot;
-        }
-        finally {
-            listLock.readLock().unlock();
-        }
-    }
-
-    public Account addAccount(Account account,int accountId)
-    {
-        if(accountId>=this.MAXACCOUNTNUMBER||accountId<0){
-            throw new IllegalArgumentException("accountId>MAXACCOUNTNUMBER || accountId<0");
-        }
         Account prev;
         this.listLock.writeLock().lock();
-        try{
-            prev=accountList[accountId];
+        try {
+            prev = accountList[accountId];
 
             //gestione length
-            if(prev!=null&&account==null){
+            if (prev != null && account == null) {
                 this.length--;
-            }else if(prev==null && account!=null){
+            } else if (prev == null && account != null) {
                 this.length++;
             }
-            if(account==null){
-                lastFreeposition=accountId;
-            }else{
+            if (account == null) {
```

Comment:
A questo punto per il commento sotto non sò se sia così utile questo controllo per settare poi lastFreePosition in quanto è improbabile che chiamiamo una addAccount(null,qualcosa)

Reply:
Ma vabbè niente lasciamola non dobbiamo intanto essere super mega performanti sulle add